### PR TITLE
CNV-23418: unsupported escape hatch mechanism custom HS/KV vms

### PR DESCRIPTION
**Pull request by nunnatsa**
Suport the new HostedCluster/NodePool annotation
"hypershift.openshift.io/kubevirt-vm-jsonpatch", to modify the KubeVirt VM template using jsonpatch string.

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

**Comment by nunnatsa** [comment]
/hold

WIP

File: hypershift-operator/controllers/nodepool/nodepool_controller.go [hunk 1]
```diff
@@ -2488,7 +2488,11 @@ func machineTemplateBuilders(hcluster *hyperv1.HostedCluster, nodePool *hyperv1.
 		}
 	case hyperv1.KubevirtPlatform:
 		template = &capikubevirt.KubevirtMachineTemplate{}
-		machineTemplateSpec = kubevirt.MachineTemplateSpec(nodePool, kubevirtBootImage, hcluster)
+		var err error
+		machineTemplateSpec, err = kubevirt.MachineTemplateSpec(nodePool, kubevirtBootImage, hcluster)
+		if err != nil {
+			return nil, nil, "", err
+		}
```
Comment (davidvossel) [inline hypershift-operator/controllers/nodepool/nodepool_controller.go:2495]: we need this error to bubble up to a condition somewhere so we have visibility that the controller is blocked due to json patch errors.

Comment (davidvossel) [reply hypershift-operator/controllers/nodepool/nodepool_controller.go:2495]: I tracked this down a bit. We do get a ReconsileError event associated with the nodepool if this fails, which I think is good enough for now. It at least means we don't have to look at a log message to find when this issue is occurring. 

```
r.recorder.Eventf(nodePool, corev1.EventTypeWarning, "ReconcileError", "%v", err)
```

**Review by davidvossel — COMMENTED** [review]
/lgtm
/approve

**Review by davidvossel — COMMENTED** [review]
/lgtm

**Pull request merged**